### PR TITLE
AB#6926 Move CTA button for plans

### PR DESCRIPTION
**Pull request by additiverich**
[ADO](https://dev.azure.com/S72/SHIFT72/_boards/board/t/Web%20team/Stories/?workitem=6926) | [Zeplin ](https://app.zeplin.io/project/607f961cf9f03790cc66e475/screen/61396196750e52b971758c75)

- move CTA button into pageHeader
- pageContent doesn't need a plan anymore
- modify page header CSS for curated plan show case pages only
- create 1rem spacing variable for vertical spacing purposes

### CODE REVIEW

File: site/templates/page/page-header.jet [hunk 1]
```diff
@@ -5,9 +5,9 @@
         <s72-plan-label data-slug="{{plan.Slug}}" data-title="{{plan.Name}}"></s72-plan-label>
       {{end}}
       <h1>{{title}}</h1>
-      {{if plan != 0 && plan.HasExpiryDate() }}
-        {{ if plan.HasExpiryDate() }}
-          <p class="alert p-0 border-0 my-1">
```
Comment (matthewthomsonnz) [inline site/templates/page/page-header.jet:10]: My opinionated take is this is a step in the wrong direction

File: site/styles/_variables.scss [hunk 2]
```diff
@@ -105,6 +105,7 @@ $spacers: (
   // 40px
   '5': $spacer * 5,
   // 50px
+  '1rem': 1rem
```
Comment (stajs) [inline site/styles/_variables.scss:108]: lel

File: site/templates/page/page-header.jet (hunk 1 above)
Comment (stajs) [reply site/templates/page/page-header.jet:10]: How so?

**Review by stajs — APPROVED** [review]
Looks fine to me, _however_  just to throw an idea: is it worth having a specific page type for plans rather than overloading curated pages?

File: site/templates/page/page-header.jet (hunk 1 above)
Comment (additiverich) [reply site/templates/page/page-header.jet:10]: Ping @matthewthomsonnz - what's the step in the wrong direction?

Comment (matthewthomsonnz) [reply site/templates/page/page-header.jet:10]:
```
.curated-plan-page {
  .page-header {
    .alert {
      @extend .p-0;
      @extend .border-0;
```
I know inline utility classes are controversial, but I think they are a good solution for one offs, and mitigate nested css hell

Comment (stajs) [reply site/templates/page/page-header.jet:10]: Ah, the removal of the utility classes are the step in the wrong direction. I'm not against utility classes per-se and `@extend` can lead to more bloat in the final CSS. In this case I agree with you, but it's pretty marginal so I don't mind either way.

Comment (stajs) [reply site/templates/page/page-header.jet:10]: PS, reason why it was hard to spot the problem, is the view on the "Files changed" tab points out the utility classes while the one in "Conversation" doesn't:

image

image

Comment (additiverich) [reply site/templates/page/page-header.jet:10]: @matthewthomsonnz Interested in your feedback now

File: site/templates/page/page-header.jet [hunk 3]
```diff
@@ -1,13 +1,13 @@
...
   {{if len(title) > 0}}
-    <div class="page-header">
+    <div class="page-header pb-0">
       {{if plan != 0}}
         <s72-plan-label data-slug="{{plan.Slug}}" data-title="{{plan.Name}}"></s72-plan-label>
       {{end}}
       <h1>{{title}}</h1>
-      {{if plan != 0 && plan.HasExpiryDate() }}
-        {{ if plan.HasExpiryDate() }}
-          <p class="alert p-0 border-0 my-1">
```
Comment (matthewthomsonnz) [inline site/templates/page/page-header.jet:10]: Re-add back in the 'alert' inline class and good to go (without it, the icon color rule [here](https://github.com/shift72/core-template/blob/develop/site/styles/_icons.scss#L35-L37) is not primary as per [design](https://app.zeplin.io/project/607f961cf9f03790cc66e475/screen/61396196750e52b971758c75))

**Pull request merged**